site-scrapers/PediatricAssociatesOfGreaterSalem.js

```diff
@@ -0,0 +1,188 @@
+    days,
+    url,
+    bearerToken,
+    schedulingToken
+) {
+    const postData = JSON.stringify({
+        operationName: "SearchAvailabilityDates",
+        variables: {
+            locationIds: ["2804-102"],
+            practitionerIds: [],
+            specialty: "Unknown Provider",
+            serviceTypeTokens: [
+                // TODO: I am not sure what this GUID signifies or if it varies.
+                "codesystem.scheduling.athena.io/servicetype.canonical|49b8e757-0345-4923-9889-a3b57f05aed2",
+            ],
+            startAfter: Today(),
+            startBefore: DaysLater(days - 1),
+        },
+        query: SearchAvailabilityDatesQuery(),
+    });
+
+    const options = {
+        method: "POST",
+        headers: {
+            authorization: `Bearer ${bearerToken}`,
+            "Content-Type": "application/json",
+            "Content-Length": postData.length,
+            "x-scheduling-jwt": schedulingToken,
+        },
+    };
+
+    const responseData = await Post(url, postData, options);
+    return JSON.parse(responseData);
+}
+
+function Today() {
+    // Today's date in format 2021-02-26T00:00:00-05:00 (ISO-8601)
+    return moment().startOf("day").format();
+}
+
+function DaysLater(days) {
+    // The last second of "days" days later e.g. 2021-03-13T23:59:59-05:00
+    return moment().add(days, "days").endOf("day").format();
+}
+
+function SearchAvailabilityDatesQuery() {
+    return `query SearchAvailabilityDates($locationIds: [String!], $practitionerIds: [String!], $specialty: String, $serviceTypeTokens: [String!]!, $startAfter: String!, $startBefore: String!, $visitType: VisitType) {
+        searchAvailabilityDates(locationIds: $locationIds, practitionerIds: $practitionerIds, specialty: $specialty, serviceTypeTokens: $serviceTypeTokens, startAfter: $startAfter, startBefore: $startBefore, visitType: $visitType) {
+            date
+            availability
+            __typename
+        }}`;
+}
+
+function Post(url, postData, options) {
+    return new Promise((resolve) => {
+        const req = https.request(url, options, (res) => {
+            let body = "";
+            res.on("data", (chunk) => {
+                body += chunk;
+            });
+            res.on("end", () => {
+                if (res.statusCode >= 200 && res.statusCode < 300) {
+                    resolve(body);
+                } else {
+                    console.error(
+                        `Error status code [${res.statusCode}] returned: ${res.statusMessage}`
+                    );
+                    resolve();
+                }
+            });
+        });
+        req.write(postData);
+        req.on("error", (e) => {
+            console.error("Error making request: " + e);
+        });
+        req.end();
+    });
+}
+
+function ParseAvailabilityDates(availabilityDates) {
+    // Expected response like
+    // [
+    //     {"date":"2021-02-26","availability":false,"__typename":"Availability"},
```

Comment:
Is there any way we can know how many slots are available on a given day?

Reply:
I had only 1 shot at observing the web request sequence unfortunately.
When this site says "no availability for 3 months" it uses a different sequence (GetFilters).
So I think there was another request after this one but I don't have any information about it.
I think once there is availability for at least 1 day we could figure it out.

Reply:
That's fine :) We'll monitor and see if we can catch one if it opens up